File: CustomToolTemplateCS/Extensions.cs
using System;
using Corel.Interop.VGCore;

public static class Extensions
{
    /// <summary>
    /// Active optimization features, and start a command group
    /// </summary>
    /// <param name="app"></param>
    /// <param name="commandGroup"></param>
    /// <param name="optimization"></param>
    /// <param name="enableEvents"></param>
    /// <param name="preservSeletion"></param>
	 public static void BeginDraw(this Corel.Interop.VGCore.Application app, bool commandGroup = true, bool optimization = true, bool enableEvents = false, bool preservSeletion = true)
        {
            if (app.ActiveDocument != null)
            {
                if (commandGroup)
                    app.ActiveDocument.BeginCommandGroup();
                if (preservSeletion)
                    app.ActiveDocument.PreserveSelection = preservSeletion;
            }

            app.Optimization = optimization;
            app.EventsEnabled = enableEvents;
        }
    /// <summary>
    /// Desables optimization features, close the command group and reflesh UI
    /// </summary>
    /// <param name="app"></param>
        public static void EndDraw(this Corel.Interop.VGCore.Application app)
        {
            if (app.ActiveDocument != null)
            {
                app.ActiveDocument.EndCommandGroup();
                app.ActiveDocument.PreserveSelection = false;
            }
            app.Optimization = false;
            app.EventsEnabled = true;
            app.Refresh();
        }
}

File: DockerTemplateVB/Extensions.vb
Imports System
Imports Corel.Interop.VGCore
Imports System.Runtime.CompilerServices

Module Extensions
    <Extension()>
    Sub BeginDraw(ByVal app As Corel.Interop.VGCore.Application, ByVal Optional commandGroup As Boolean = True, ByVal Optional optimization As Boolean = True, ByVal Optional enableEvents As Boolean = False, ByVal Optional preservSeletion As Boolean = True)
        If app.ActiveDocument IsNot Nothing Then
            If commandGroup Then app.ActiveDocument.BeginCommandGroup()
            If preservSeletion Then app.ActiveDocument.PreserveSelection = preservSeletion
        End If

        app.Optimization = optimization
        app.EventsEnabled = enableEvents
    End Sub

    <Extension()>
    Sub EndDraw(ByVal app As Corel.Interop.VGCore.Application)
        If app.ActiveDocument IsNot Nothing Then
            app.ActiveDocument.EndCommandGroup()
            app.ActiveDocument.PreserveSelection = False
        End If

        app.Optimization = False
        app.EventsEnabled = True
        app.Refresh()
    End Sub
End Module
